sdk/helper/ldaputil/client.go

```diff
@@ -400,7 +400,7 @@ func (c *Client) performLdapFilterGroupsSearchPaging(cfg *ConfigEntry, conn Pagi
 			cfg.GroupAttr,
 		},
 		SizeLimit: math.MaxInt32,
-	}, math.MaxInt32)
+	}, uint32(cfg.MaximumPageSize))
```

Comment:
Wondering if you were able to test that setting this to 0 works with an LDAP server that does not permit paging? Just want to make sure that the reporter's issue is solved by setting this to 0.

Slightly related to that, do you know under which pre-condition we'd enter the branch at [client.go#L525](https://github.com/hashicorp/vault/blob/main/sdk/helper/ldaputil/client.go#L525)? I tried to find where a connection would satisfy `PagingConnection` or `Connection`, but it's not super clear. I wonder if we're always landing on a `PagingConnection` and can delete the other code path.

Reply:
> Wondering if you were able to test that setting this to 0 works with an LDAP server that does not permit paging? Just want to make sure that the reporter's issue is solved by setting this to 0.

I am having some trouble setting up a dev LDAP instance that doesn't support paging to actually confirm this, but I will keep chugging along at it.

>Slightly related to that, do you know under which pre-condition we'd enter the branch at [client.go#L525](https://github.com/hashicorp/vault/blob/main/sdk/helper/ldaputil/client.go#L525)? I tried to find where a connection would satisfy PagingConnection or Connection, but it's not super clear. I wonder if we're always landing on a PagingConnection and can delete the other code path.

My thinking with this was since this is a function that is part of the SDK, some other process outside of the core Vault code could be passing in a `Connection` into `GetLdapGroups` and wouldn't necessarily have updated to using a `PagingConnection`.  But you are correct, in terms of the Vault codebase, we would never actually reach that path.